addons/web/static/src/views/pivot/pivot_group_by_menu.js

```diff
@@ -44,16 +44,20 @@ export class PivotGroupByMenu extends GroupByMenu {
     }
     /**
      * @override
-     * @param {CustomEvent} ev
+     * @param {Object} param0
+     * @param {number} param0.itemId
+     * @param {number} [param0.optionId]
      */
-    onGroupBySelected(ev) {
-        const { itemId, optionId } = ev.detail.payload;
-        const item = this.items.find(({ id }) => id === itemId);
-        // Augment payload data
-        Object.assign(ev.detail.payload, {
-            fieldName: item.fieldName,
-            interval: optionId,
-            groupId: this.props.cell.groupId,
-        });
+    onGroupBySelected({ itemId, optionId }) {
+        if (this.props.onItemSelected) {
```

Comment:
isn't the prop mandatory (a fortiori for the PivotGroupByMenu)?

Reply:
Good point. I've made it mandatory through props validation.